src/new-client/src/services/connection.ts

```diff
@@ -25,3 +26,25 @@ export const [useConnectionStatus, connectionStatus$] = bind(
   hConnectionStatus$().pipe(map(connectionMapper)),
   ConnectionStatus.CONNECTING,
 )
+
+export const withIsStaleData = <T>(
+  source$: Observable<T>,
+): Observable<boolean> => {
+  const sourceDate$ = source$.pipe(
+    map(() => Date.now()),
+    startWith(0),
+  )
+
+  const connectionDate$ = connectionStatus$.pipe(
+    filter((status) => status !== ConnectionStatus.CONNECTING),
+    map((status) => ({ status, date: Date.now() })),
+  )
+
+  return combineLatest([sourceDate$, connectionDate$]).pipe(
+    map(([sourceDate, connection]) => {
+      if (connection.status !== ConnectionStatus.CONNECTED) return false
```

Comment:
If we only care about the CONNECTED status, can this be lifted into the filter above?

Reply:
Good catch! Thanks!